ultralytics/engine/trainer.py

```diff
@@ -225,9 +225,9 @@ def _setup_ddp(self, world_size):
         torch.cuda.set_device(RANK)
         self.device = torch.device("cuda", RANK)
         # LOGGER.info(f'DDP info: RANK {RANK}, WORLD_SIZE {world_size}, DEVICE {self.device}')
-        os.environ["NCCL_BLOCKING_WAIT"] = "1"  # set to enforce timeout
+        os.environ["TORCH_NCCL_BLOCKING_WAIT"] = "1"  # set to enforce timeout
         dist.init_process_group(
-            "nccl" if dist.is_nccl_available() else "gloo",
+            backend="nccl" if dist.is_nccl_available() else "gloo",
```

Comment:
Could you please revert these changes that are less relevant to this issue?

Reply:
ok. those changes were in the newer ultralytics repo and I saw some warnings and at first wasn't sure if they also had something to do with the issue. I guess not.